### PR TITLE
Use a temporary CodaLabManager for WorkerManagers

### LINKED ISSUE #3078: CLI frequently forgets which worksheet I am on
**Describe the bug**
After not working for a few minutes, codalab forgets the result of the previous "cl work nlp::" command and tries to connect to the public site.

**Codalab Information**
- *Your username*:   AngledLuffa
- *Error message received from Codalab (if applicable)*:
Requesting access at https://worksheets.codalab.org
 - *OS (if on CLI)*:
this is on the sc server on the stanford cluster
 - *Codalab CLI Version (if on CLI)*: 0.5.29

**To Reproduce**
Steps to reproduce the behavior :
1. Do codalab stuff
2. Wait a little while (much less than an hour)
3. have to run "cl work nlp::" again

**Expected behavior**
Would prefer it to stay on the same worksheet

### PR DESCRIPTION
**Pull request by teetone**
### Reasons for making this change

The changes in this PR is an alternative way to fix the sessions issue. A temporary `CodaLabManager` maintains its state in memory and never writes to `state.json`. Therefore, a WorkerManager in a different screen session wouldn't be able to erase sessions information stored in `state.json`.

### Related issues

Related PR: https://github.com/codalab/codalab-worksheets/pull/3145
Resolves https://github.com/codalab/codalab-worksheets/issues/3078

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by epicfaace** [comment]
Are you sure this solves #3078 in all cases? If a user has multiple sessions open (without a session created by a WorkerManager), I think this issue would still happen, right?

**Comment by teetone** [comment]
> Are you sure this solves #3078 in all cases? If a user has multiple sessions open (without a session created by a WorkerManager), I think this issue would still happen, right?

We synced about this in our meeting, but this will only fix the worker manager problem.

File: codalab/worker_manager/worker_manager.py
Comment (epicfaace) [inline codalab/worker_manager/worker_manager.py:92]: For clarity

```suggestion
        self.codalab_manager = CodaLabManager(temporary=args.temp_session)
```

**Review by epicfaace — APPROVED** [review]

**Comment by teetone** [comment]
I will wait for @nelson-liu's approval too just in case.

**Review by nelson-liu — APPROVED** [review]

**Pull request merged**
